ListLibrary/MyArrayList.cs

```diff
@@ -0,0 +1,532 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ListLibrary
+{
+    public class MyArrayList<T> : IMyList<T> where T : IComparable<T>
+    {
+        private const int _defaultCapacity = 4;
+        private const double _capacityRise = 1.33;
+        private T[] _items;
+        private int _size;
+
+        public int Count => _size;
+        public int Capacity => _items.Length;
+        public T this[int index]
+        {
+            get
+            {
+                if (index >= Count || index < 0)
+                {
+                    throw new ArgumentException("Index should be less than count and more than zero");
+                }
+
+                return _items[index];
+            }
+
+            set
+            {
+                if (index >= Count || index < 0)
+                {
+                    throw new ArgumentException("Index should be less than count and more than zero");
+                }
+
+                _items[index] = value;
+            }
+        }
+
+        #region Constructors and Interfaces implementation
+
+        public MyArrayList()
+        {
+            _items = new T[_defaultCapacity];
+        }
+
+        public MyArrayList(T element)
+        {
+            if (element == null)
+            {
+                throw new ArgumentException("Element can't be null");
+            }
+
+            _items = new T[_defaultCapacity];
+            Add(element);
+        }
+
+        public MyArrayList(IEnumerable<T> collection)
+        {
+            if (collection == null)
+            {
+                throw new ArgumentException("Elements can't be null");
+            }
+
+            _items = new T[_defaultCapacity];
```

Comment:
You could have here a lot of adds. Use `ToArray` or `ToList` on source collection to get count and then create array with valid size